apps/oxlint/src-js/plugins/source_code.ts

```diff
@@ -169,29 +169,66 @@ export const SOURCE_CODE = Object.freeze({
    * @param nodeOrToken - The AST node or token to check for adjacent comment tokens.
    * @returns Array of `Comment`s in occurrence order.
    */
-  // oxlint-disable-next-line no-unused-vars
   getCommentsBefore(nodeOrToken: NodeOrToken): Comment[] {
-    throw new Error('`sourceCode.getCommentsBefore` not implemented yet'); // TODO
+    if (ast === null) initAst();
+    const { comments } = ast;
+    if (comments.length === 0) return [];
+    const targetStart = nodeOrToken.range[0];
+    let i = comments.length - 1;
+    // linear backwards search for the index of the last comment before the provided node or token, TODO(perf): binary
+    while (i >= 0 && comments[i].range[1] > targetStart) {
+      i--;
+    }
+    return comments.slice(0, i + 1);
   },
 
   /**
    * Get all comment tokens directly after the given node or token.
    * @param nodeOrToken - The AST node or token to check for adjacent comment tokens.
    * @returns Array of `Comment`s in occurrence order.
    */
-  // oxlint-disable-next-line no-unused-vars
   getCommentsAfter(nodeOrToken: NodeOrToken): Comment[] {
-    throw new Error('`sourceCode.getCommentsAfter` not implemented yet'); // TODO
+    if (ast === null) initAst();
+    const { comments } = ast;
+    if (comments.length === 0) return [];
+    const targetEnd = nodeOrToken.range[1];
+    let i = 0;
+    // linear forwards search for the index of the first comment after the provided node or token, TODO(perf): binary
+    while (i < comments.length && comments[i].range[0] < targetEnd) {
```

Comment:
```suggestion
    const len = comments.length;
    while (i < len && comments[i].range[0] < targetEnd) {
```

...to avoid `comments.length` property lookup on each turn of the loop.